src/builtins/Transformers.jl

```diff
@@ -1022,14 +1022,58 @@ function MMI.transform(transformer::ContinuousEncoder, fitresult, X)
 end
 
 
+# # INTERACTION TRANSFORMER
+
+@with_kw_noshow mutable struct InteractionTransformer <: Static
+    order::Int                                          = 2
+    colnames::Union{Nothing, Vector{Symbol}}            = nothing
+    function InteractionTransformer(order, colnames)
+        order >= 2 || throw(ArgumentError("The order of interactions should be at least 2."))
+        if colnames !== nothing
+            length(colnames) > 1 || throw(ArgumentError("At least 2 columns are required to compute interactions."))
+        end
+        return new(order, colnames)
+    end
+end
+
```

Comment:
I suggest you put the hyperparameter checks in a `clean!` method and not have an internal constructor. The `clean!` method gets called at every `fit!`, so if user mutates hyperparameters illegally, this is caught. See eg, [here](https://github.com/JuliaAI/MLJTuning.jl/blob/e7e54bd14a03ff6179402f8d9eb918b018492870/src/tuned_models.jl#L360). 

Instead of `@with_kw_noshow` you can use `@mlj_model` (see [here](https://alan-turing-institute.github.io/MLJ.jl/dev/adding_models_for_general_use/#Macro-shortcut)) and then that will interpolate a `clean!` call into the generated keyword argument constructor as well.

Reply:
Thanks for that, I have changed in favor of the `@mlj_model` macro as suggested